ppx/melange_ppx.ml

```diff
@@ -616,9 +613,36 @@ module Mapper = struct
                         pval_attributes = [];
                         pval_prim =
                           Melange_ffi.External_ffi_types.inline_string_primitive
-                            s dec;
+                            s None;
                       };
                 }
+            | ( Some ({ attr_name = { txt; loc }; _ } as attr),
+                Pexp_constant (Pconst_string (s, _, Some dec)) ) -> (
+                match
+                  Melange_ffi.Utf8_string.Interp.transform ~loc ~delim:dec
```

Comment:
Why does the transform have to be moved from the expression to the structure item section?

Reply:
That's not what's happening here. This code concerns only `[@@mel.inline]`, which transforms `let x = {j| ... |j} [@@mel.inline]` into an external with `Ffi_inline_const (Const_string {s;unicode})`.

We perform this transformation in the PPX because there's code that can be inlined, and code that can't. Example:

```ocaml
let x = {j| foo |j} [@@mel.inline] (* <-- constant string, can be inlined *)

(* will be transformed into `"foo" ^ x`, which isn't a constant, and cannot be inlined *)
let y = {j| foo $(x) |j} [@@mel.inline] 
```

Reply:
> That's not what's happening here.

I am confused. This code seems to have been moved from line 405, where it was inside the `method! expression e` section, into this section, which is the `method! structure_item str` one.

> We perform this transformation in the PPX because there's code that can be inlined, and code that can't.

Makes sense.

Reply:
It’s moved to structure item to catch the only the inlined cases. The remaining transformations happen in the core lib now